lib/src/react_client/event_helpers.dart

```diff
@@ -564,61 +736,76 @@ SyntheticWheelEvent createSyntheticWheelEvent({
   num deltaY,
   num deltaZ,
 }) {
-  return SyntheticWheelEvent(
-    bubbles ?? baseEvent?.bubbles ?? false,
-    cancelable ?? baseEvent?.cancelable ?? true,
-    currentTarget ?? baseEvent?.currentTarget,
-    defaultPrevented ?? baseEvent?.defaultPrevented ?? false,
-    preventDefault ?? baseEvent?.preventDefault ?? () {},
-    stopPropagation ?? baseEvent?.stopPropagation ?? () {},
-    eventPhase ?? baseEvent?.eventPhase,
-    isTrusted ?? baseEvent?.isTrusted ?? false,
-    nativeEvent ?? baseEvent?.nativeEvent,
-    target ?? baseEvent?.target,
-    timeStamp ?? baseEvent?.timeStamp ?? 0,
-    type ?? baseEvent?.type ?? 'empty event',
-    deltaX ?? baseEvent?.deltaX,
-    deltaMode ?? baseEvent?.deltaMode,
-    deltaY ?? baseEvent?.deltaY,
-    deltaZ ?? baseEvent?.deltaZ,
-  );
+  return jsifyAndAllowInterop({
+    ..._wrapBaseEventPropertiesInMap(
+      baseEvent: baseEvent,
+      bubbles: bubbles,
+      cancelable: cancelable,
+      currentTarget: currentTarget,
+      defaultPrevented: defaultPrevented,
+      preventDefault: preventDefault,
+      stopPropagation: stopPropagation,
+      eventPhase: eventPhase,
+      isTrusted: isTrusted,
+      nativeEvent: nativeEvent,
+      target: target,
+      timeStamp: timeStamp,
+      type: type,
+    ),
+    'deltaX': deltaX ?? baseEvent?.deltaX,
+    'deltaMode': deltaMode ?? baseEvent?.deltaMode,
+    'deltaY': deltaY ?? baseEvent?.deltaY,
+    'deltaZ': deltaZ ?? baseEvent?.deltaZ,
+  }) as SyntheticWheelEvent;
 }
 
 extension SyntheticEventTypeHelpers on SyntheticEvent {
-  /// Returns whether this is a [SyntheticClipboardEvent].
-  bool get isClipboardEvent => this is SyntheticClipboardEvent;
+  bool _checkEventType(List<String> types) => type != null && types.any((t) => type.contains(t));
 
-  /// Returns whether this is a [SyntheticKeyboardEvent].
-  bool get isKeyboardEvent => this is SyntheticKeyboardEvent;
+  /// Whether the event instance has been removed from the ReactJS event pool.
+  ///
+  /// > See: [persist]
+  @Deprecated('The modern event system does not use pooling. This always returns true.')
```

Comment:
Can we put a version number in here telling the consumer when we plan to remove this?